Added dynamic battery widget

RaresCon opened the pull request:
## Description

For bottom to know that there are no batteries on the system, I added the `battery::Manager` to the options.rs file because here is the first moment bottom verifies battery configuration by reading the config file, which may or may not contain the battery field, but for a better UX, it doesn't matter what bottom finds in the config file now, if it doesn't retrieve battery data, it just ignores the battery widget all together.
If needed, it can be adjusted so that if the config file contains the battery field, it will still show the widget even without batteries on the system.

## Issue

#1  

## Testing

_If relevant, please state how this was tested. All changes **must** be tested to work:_
It was tested with and without config files on a PC without battery and on a laptop with and without battery installed.

_If this is a code change, please also indicate which platforms were tested:_

- [x] _Windows_
- [x] _macOS_
- [x] _Linux_

## Checklist

_If relevant, ensure the following have been met:_

- [x] _Areas your change affects have been linted using rustfmt (`cargo fmt`)_
- [x] _The change has been tested and doesn't appear to cause any unintended breakage_
- [x] _Documentation has been added/updated if needed (`README.md`, help menu, doc pages, etc.)_
- [ ] _The pull request passes the provided CI pipeline_
- [x] _There are no merge conflicts_
- [ ] _If relevant, new tests were added (don't worry too much about coverage)_

Commit:
Added dynamic battery widget

For bottom to know that there are no batteries on the system,
I added the battery::Manager to the options.rs file because
here is the first moment bottom verifies battery configuration
by reading the config file, which may or may not contain the
battery field, but for a better UX, it doesn't matter what bottom
finds in the config file now, if it doesn't retrieve battery data,
it just ignores the battery widget all together.
If needed, it can be adjusted so that if the config file contains
the battery field, it will still show the widget.

alexandruradovici commented:
I tested this using macOS and it works.

alexandruradovici commented on src/options.rs:
```diff
@@ -10,6 +10,7 @@ use clap::ArgMatches;
 use layout_options::*;
 use regex::Regex;
 use serde::{Deserialize, Serialize};
+use starship_battery::Manager;
```
This fails if the battery features is not turned on, please guard it with a `#cfg` directive.

RaresCon replied:
I added the guarding using another account unfortunately (NitroDev), which is another account of mine. It should be all working fine now. If you want, I can redo this pushed commit with this account (RaresCon).

NitrogenDev committed:
CFG guarding for BATTERY module

I guarded the options.rs in two places for battery module that can be missing in the feature list.

alexandruradovici commented:
I think you can send the PR to upstream, place a link to it in the pull request.

RaresCon commented:
> I think you can send the PR to upstream, place a link to it in the pull request.

I've sent the PR, I've put a link to this one in there. Thank you for the tip about #cfg directive.

alexandruradovici commented:
Link https://github.com/ClementTsang/bottom/pull/975

RaresCon commented:
> Link [ClementTsang#975](https://github.com/ClementTsang/bottom/pull/975)

The PR was approved. Thank you once again!

alexandruradovici commented:
Thanks.
